### PR TITLE
Add delay_filter module

### PR DESCRIPTION
**Pull request by jsdillon**
This gets us up and running with a basic interface to `uvtools.dspec.delay_filter` that should enable simple interactive delay filtering of data and thin scripts for delay filtering files as part of the pipeline.

Also addresses #157.

### CODE REVIEW

**Comment by jsdillon** [comment]
Also, @zakiali if you have any idea how to eliminate my doubled commits, let me know.

**Comment by jsdillon** [comment]
Nevermind, I figured out how to squash commits with a rebase.

File: hera_cal/delay_filter.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                or list of UVData instances to concatenate into a single internal DataContainer
+            format: file format of data. Default 'miriad.' Ignored if input_data is UVData object(s).
+        '''
+        if isinstance(input_data, (str,UVData)):
+            self.writable = True
+            self.input_data, self.format = input_data, format
+        self.data, self.flags, self.antpos, _, self.freqs, self.times, _, _ = io.load_vis(input_data, return_meta=True, format=format)
+
+
+    def load_dicts(self, data, flags, freqs, antpos):
```
Comment (zakiali) [inline hera_cal/delay_filter.py]: Was there a reason you split the load_data and load_dicts function? Can we combine them?

File: hera_cal/delay_filter.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    def __init__(self):
+        '''Class for loading data, performing uvtools.dspec.delay_filter, and writing out data using pyuvdata.
+        To use, run either self.load_data() or self.load_dicts(), then self.run_filter(). If data is loaded with a single 
+        string path or a single UVData object, it can be written to a new file using self.write_filtered_data().
+        '''
+        self.writable = False
+
+
+    def load_data(self, input_data, format='miriad'):
```
Comment (zakiali) [inline hera_cal/delay_filter.py]: Can we change "format" to "file_type". format is a python builtin and we use file_type in other places.

File: hera_cal/delay_filter.py (hunk 1 above)
Comment (zakiali) [reply hera_cal/delay_filter.py]: rename function to load_data_dicts. Makes it a bit clear.

File: hera_cal/delay_filter.py (hunk 2 above)
Comment (jsdillon) [reply hera_cal/delay_filter.py]: I'm using filetype (no underscore, which matches abscal and firstcal).

File: hera_cal/delay_filter.py (hunk 1 above)
Comment (jsdillon) [reply hera_cal/delay_filter.py]: I'm renaming load_dicts to load_data_as_dicts to be more clear.

**Pull request merged**